CMakeLists.txt

```diff
@@ -16,11 +22,66 @@ if(ALPAKA_ACC_GPU_CUDA_ENABLE)
     set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};--expt-extended-lambda")
 endif()
 
-# Use doxygen for api doc when available.
-find_package(Doxygen QUIET)
-if(DOXYGEN_FOUND)
+
+add_library(${PROJECT_NAME} INTERFACE)
+target_include_directories(${PROJECT_NAME}
+  INTERFACE
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
+target_link_libraries(${PROJECT_NAME}
+  INTERFACE
+  alpaka::alpaka)
+add_library(vikunja::vikunja ALIAS ${PROJECT_NAME})
+
+
+if(${CMAKE_PROJECT_NAME} STREQUAL ${PROJECT_NAME})
+  #########################################
+  ######### Install Configuration #########
+  #########################################
+
+  include(CMakePackageConfigHelpers)
+  write_basic_package_version_file(
+    "${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+    VERSION 0.1
+    COMPATIBILITY AnyNewerVersion
+    )
+
+  include(GNUInstallDirs)
+  install(TARGETS ${PROJECT_NAME} alpaka
+    EXPORT "${PROJECT_NAME}Targets"
+    )
+
+  include(CMakePackageConfigHelpers)
+  configure_package_config_file(
+    "${PROJECT_SOURCE_DIR}/cmake/${PROJECT_NAME}Config.cmake.in"
+    "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
+    )
+
+  install(EXPORT "${PROJECT_NAME}Targets"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
+    NAMESPACE ${PROJECT_NAME}::)
+  install(FILES
+    "${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+    "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+  install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/ DESTINATION include)
+
+  #########################################
+  ############# Documentation #############
+  #########################################
+
+  # Use doxygen for api doc when available.
+  find_package(Doxygen QUIET)
+  if(DOXYGEN_FOUND)
     doxygen_add_docs(doc include/)
+  endif()
```

Comment:
There was a talk on CMake at CppCon by Robert Schumacher (https://www.youtube.com/watch?v=sBP17HQAQjk) where he complained that some libraries take longer to build whether or not doxygen is installed or now. He recommended to at least give an option to disable building the documentation. Personally, I would not even build the documentation by default.

Reply:
I made it optional